ruskeyverbs/views.py

```diff
@@ -26,19 +51,618 @@ def index(request):
+                             """Something went wrong.
+                             You did not enter a valid answer.""")
+            return HttpResponseRedirect(reverse('fill-in-the-blank',
+                                                args=[pk]))
+    else:
+        raise Http404
+
+
+@login_required
+def ArrangeWords(request, pk):
+    """
+    quiz view for the arrange words type quiz
+    """
+    example_inst = get_object_or_404(Example, pk=pk)
+    """
+    create a randomized list of the words in the sentence, which will be the
+    options for the choice fields in the arrange words form
+    """
+    randomized_russian_text_list = strip_punct_lower(example_inst.russian_text).split()
+    sentence_len = len(randomized_russian_text_list)
+    """
+    add three other random forms the verb under quiz (that are not already in
+    the sentence) to the list
+    """
+    randomized_verb_forms = example_inst.verb.get_forms_list(stressed=False,
+                                                             random=True)
+    shuffle(randomized_verb_forms)
+    count = 0
+    for verb_form in randomized_verb_forms:
+        if verb_form not in randomized_russian_text_list:
+            randomized_russian_text_list.append(verb_form)
+            count += 1
+        if count >= 3:
+            break
+    request.session['extra_options'] = randomized_russian_text_list
+    # create a list of tuples that will be the choices for the choice fields
+    shuffle(randomized_russian_text_list)
+    tuple_list = []
+    for i in range(len(randomized_russian_text_list)):
+        tuple_list.append((randomized_russian_text_list[i],
+                           randomized_russian_text_list[i]))
+    form = ArrangeWordsForm(tuple_list, sentence_len)
+    try:
+        request.session['quiz_state'] += 1
+        request.session.modified = True
+    except KeyError:
+        request.session['quiz_state'] = 0
+    if request.session['quiz_state'] == 3:
+        return render(request,
+                      'ruskeyverbs/arrange_words.html',
+                      context={'russian_text': example_inst.russian_text,
+                               'english_text': example_inst.translation_text,
+                               'form': form,
+                               'pk': pk,
+                               'file': example_inst.example_audio})
+    else:
+        raise Http404
+
+
+@login_required
+def ArrangeWordsEval(request, pk):
+    """
+    eval view for arrange words quiz
+    """
+    example_inst = get_object_or_404(Example, pk=pk)
+    """
+    create a list of the words in the sentence, which will be used
+    to bind the form
+    """
+    russian_text_list = strip_punct_lower(example_inst.russian_text,
+                                          stressed=True).split()
+    russian_text_list_no_stress = strip_punct_lower(
+        example_inst.russian_text).split()
+    tuple_list = []
+    for verb_form in request.session['extra_options']:
```

Comment:
Why do you need a session here? Maybe a hidden form input would be a better way to persist some of these values?

Reply:
Here too, I was using sessions as a way of ensuring consistency with randomly generated content across a quiz and an evaluation view. Do you think using a hidden form input could achieve the same result in a more efficient manner?

Reply:
Either is fine I guess. Session is fine.